tests/integration/pilot/ingress_test.go

```diff
@@ -383,175 +384,191 @@ spec:
 					name: "https-foo",
-					call: echo.CallOptions{
-						Port: echo.Port{
-							Protocol: protocol.HTTPS,
-						},
-						HTTP: echo.HTTP{
-							Path:    "/test",
-							Headers: headers.New().WithHost("foo.example.com").Build(),
-						},
-						TLS: echo.TLS{
-							CaCert: ingressutil.IngressCredentialA.CaCert,
-						},
-						Check: successChecker,
-						Count: count,
+					call: func(ingr ingress.Instance) echo.CallOptions {
+						return echo.CallOptions{
+							Port: echo.Port{
+								Protocol: protocol.HTTPS,
+							},
+							HTTP: echo.HTTP{
+								Path:    "/test",
+								Headers: headers.New().WithHost("foo.example.com").Build(),
+							},
+							TLS: echo.TLS{
+								CaCert: ingressutil.IngressCredentialA.CaCert,
+							},
+							Check: check.And(check.OK(), check.ReachedClusters(t, ingr.Cluster(), apps.B.Clusters())),
+							Count: count,
+						}
 					},
 					path:       "/test",
 					prefixPath: "/prefix",
 				},
 				{
 					// Basic HTTPS call for bar. CaCert matches the secret
 					name: "https-bar",
-					call: echo.CallOptions{
-						Port: echo.Port{
-							Protocol: protocol.HTTPS,
-						},
-						HTTP: echo.HTTP{
-							Path:    "/test",
-							Headers: headers.New().WithHost("bar.example.com").Build(),
-						},
-						TLS: echo.TLS{
-							CaCert: ingressutil.IngressCredentialB.CaCert,
-						},
-						Check: successChecker,
-						Count: count,
+					call: func(ingr ingress.Instance) echo.CallOptions {
+						return echo.CallOptions{
+							Port: echo.Port{
+								Protocol: protocol.HTTPS,
+							},
+							HTTP: echo.HTTP{
+								Path:    "/test",
+								Headers: headers.New().WithHost("bar.example.com").Build(),
+							},
+							TLS: echo.TLS{
+								CaCert: ingressutil.IngressCredentialB.CaCert,
+							},
+							Check: check.And(check.OK(), check.ReachedClusters(t, ingr.Cluster(), apps.B.Clusters())),
+							Count: count,
+						}
 					},
 					path:       "/test",
 					prefixPath: "/prefix",
 				},
 				{
 					// HTTPS call for bar with namedport route. CaCert matches the secret
 					name: "https-namedport",
-					call: echo.CallOptions{
-						Port: echo.Port{
-							Protocol: protocol.HTTPS,
-						},
-						HTTP: echo.HTTP{
-							Path:    "/test/namedport",
-							Headers: headers.New().WithHost("bar.example.com").Build(),
-						},
-						TLS: echo.TLS{
-							CaCert: ingressutil.IngressCredentialB.CaCert,
-						},
-						Check: successChecker,
-						Count: count,
+					call: func(ingr ingress.Instance) echo.CallOptions {
+						return echo.CallOptions{
+							Port: echo.Port{
+								Protocol: protocol.HTTPS,
+							},
+							HTTP: echo.HTTP{
+								Path:    "/test/namedport",
+								Headers: headers.New().WithHost("bar.example.com").Build(),
+							},
+							TLS: echo.TLS{
+								CaCert: ingressutil.IngressCredentialB.CaCert,
+							},
+							Check: check.And(check.OK(), check.ReachedClusters(t, ingr.Cluster(), apps.B.Clusters())),
```

Comment:
It seems like this will break the 'agnostic' tests?

Is there some way we can pipe through more info into the check interface, like the opts and source?

Note opts is not actually enough in this particular case, since we have no `To` - the `apps.B` could never be inferred.

Note sure all that will be worth  the complexity... cc @stevenctl

Reply:
I considered that and I'm not against it. I didn't make the change here for 2 reasons:

1. the complexity of the change required
2. doing it this way allows some minor optimization so that we don't have to create the network map on every call.

Definitely open to making the bigger change, if everyone agrees that it's the right approach. @stevenctl WDYT?

Reply:
TrafficTestCase should already support this, if it's possible to use that here:

https://github.com/istio/istio/blob/11a37c6b029fb2e775ecde8c2e7314c36832b201/tests/integration/pilot/common/traffic.go#L68